controllers/sonataflowclusterplatform_controller.go

```diff
@@ -0,0 +1,175 @@
+			}
+			r.Recorder.Event(&instance, corev1.EventTypeWarning, "Failed", fmt.Sprintf("Failed to update SonataFlowClusterPlaform: %s", err))
+			return reconcile.Result{}, err
+		}
+
+		if target != nil {
+			target.Status.ObservedGeneration = instance.Generation
+
+			if err := r.Client.Status().Patch(ctx, target, ctrl.MergeFrom(&instance)); err != nil {
+				r.Recorder.Event(&instance, corev1.EventTypeNormal, "Status Updated", fmt.Sprintf("Updated cluster platform condition %s", instance.Status.GetTopLevelCondition()))
+				return reconcile.Result{}, err
+			}
+		}
+
+		// handle one action at time so the resource
+		// is always at its latest state
+		r.Recorder.Event(&instance, corev1.EventTypeNormal, "Updated", fmt.Sprintf("Updated cluster platform condition to  %s", instance.Status.GetTopLevelCondition()))
+
+		if target != nil && target.Status.IsReady() {
+			return reconcile.Result{}, nil
+		}
+
+		// Requeue
+		return reconcile.Result{
+			RequeueAfter: 5 * time.Second,
+		}, nil
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *SonataFlowClusterPlatformReconciler) SetupWithManager(mgr ctrlrun.Manager) error {
+	return ctrlrun.NewControllerManagedBy(mgr).
+		For(&operatorapi.SonataFlowClusterPlatform{}).
+		Watches(&operatorapi.SonataFlowPlatform{}, handler.EnqueueRequestsFromMapFunc(r.mapPlatformToClusterPlatformRequests)).
+		Watches(&operatorapi.SonataFlowClusterPlatform{}, handler.EnqueueRequestsFromMapFunc(r.mapClusterPlatformToClusterPlatformRequests)).
+		Complete(r)
+}
+
+// if actively referenced sonataflowplatform object is changed, reconcile the active SonataFlowClusterPlatform.
+func (r *SonataFlowClusterPlatformReconciler) mapPlatformToClusterPlatformRequests(ctx context.Context, object client.Object) []reconcile.Request {
+	platform := object.(*operatorapi.SonataFlowPlatform)
+	sfcPlatform, err := clusterplatform.GetActiveClusterPlatform(ctx, r.Client)
+	if err != nil && !errors.IsNotFound(err) {
+		klog.V(log.E).ErrorS(err, "Failed to get active SonataFlowClusterPlatform")
+		return nil
+	}
+
+	if sfcPlatform != nil {
+		sfpcRefNsName := types.NamespacedName{Namespace: sfcPlatform.Spec.PlatformRef.Namespace, Name: sfcPlatform.Spec.PlatformRef.Name}
+		if client.ObjectKeyFromObject(platform) == sfpcRefNsName {
+			return []reconcile.Request{{NamespacedName: client.ObjectKeyFromObject(sfcPlatform)}}
+		}
+	}
+	return nil
+}
+
+// if active sonataflowclusterplatform is changed, reconcile other SonataFlowClusterPlatforms.
+func (r *SonataFlowClusterPlatformReconciler) mapClusterPlatformToClusterPlatformRequests(ctx context.Context, object client.Object) []reconcile.Request {
+	sfcPlatform := object.(*operatorapi.SonataFlowClusterPlatform)
+	var requests []reconcile.Request
+	if sfcPlatform != nil && clusterplatform.IsActive(sfcPlatform) {
+		var scpList operatorapi.SonataFlowClusterPlatformList
+		if err := r.List(ctx, &scpList); err != nil {
```

Comment:
nit: you could use a field selector to filter out the result list and remove the need for the check. Also, you wouldn't need to check for the namespace where the object resides since it's a cluster scoped object. Just the name would suffice.
```
		fs, err := fields.ParseSelector(fmt.Sprintf("metadata.name!=%s", sfcPlatform.Name))
		if err != nil {
			klog.V(log.E).ErrorS(err, "Failed to parse field selector for cluster platform CR %s", sfcPlatform.Name)
			return nil
		}
		if err := r.List(ctx, &scpList, &ctrl.ListOptions{FieldSelector: fs}); err != nil {
		...
		}
		for _, cPlatform := range scpList.Items {
			requests = append(requests, reconcile.Request{NamespacedName: client.ObjectKeyFromObject(&cPlatform)})
		}
```

Reply:
in this case we actually want to retrieve all `SonataFlowClusterPlatforms` in the cluster. this is pertinent in the event of a delete event, where an existing "inactive" clusterplatform could become "active".

Reply:
ack.